dozer/sources/RedditSource.py

```diff
@@ -0,0 +1,229 @@
+    short_name = "reddit"
+    base_url = "https://reddit.com/"
+    description = "Gets latest posts from any given subreddit"
+
+    token_url = "https://www.reddit.com/api/v1/access_token"
+    api_url = "https://oauth.reddit.com/"
+    backup_api_url = "https://reddit.com/"
+    color = discord.Color.from_rgb(255, 69, 0)
+
+    class SubReddit(DataBasedSource.DataPoint):
+        """Represents a single subreddit with associated detail"""
+        def __init__(self, name, url, color):
+            super().__init__(name, url)
+            self.name = name
+            self.url = url
+            self.color = color
+
+    def __init__(self, aiohttp_session, bot):
+        super().__init__(aiohttp_session, bot)
+        self.access_token = None
+        self.expiry_time = None
+        self.oauth_disabled = False
+        self.subreddits = {}
+        self.seen_posts = set()
+
+    async def get_token(self):
+        """Using OAuth2, get a reddit bearer token. If this fails, fallback to non-oauth API"""
+        client_id = self.bot.config['news']['reddit']['client_id']
+        client_secret = self.bot.config['news']['reddit']['client_secret']
+        params = {
+            'grant_type': 'client_credentials'
+        }
+        auth = aiohttp.BasicAuth(client_id, client_secret)
+        response = await self.http_session.post(self.token_url, params=params, auth=auth)
+        response = await response.json()
+        try:
+            self.access_token = response['access_token']
+        except KeyError:
+            DOZER_LOGGER.critical(f"Error in {self.full_name} Token Get: {response['message']}. Switching to "
+                                  f"non-OAuth API")
+            self.oauth_disabled = True
+            return
+
+        expiry_seconds = response['expires_in']
+        time_delta = datetime.timedelta(seconds=expiry_seconds)
+        self.expiry_time = datetime.datetime.now() + time_delta
+
+    async def request(self, url, *args, headers=None, **kwargs):
+        """Make a request using OAuth2 (or not, if it's been disabled)"""
+        if datetime.datetime.now() > self.expiry_time:
+            DOZER_LOGGER.info("Refreshing Reddit token due to expiry time")
+            await self.get_token()
+
+        if headers is None:
+            headers = {}
+        headers['Authorization'] = f"Bearer {self.access_token}"
+        headers['User-Agent'] = "Dozer/Discord by /u/bkeeneykid"
+
+        if self.oauth_disabled:
+            url = f"{self.backup_api_url}/{url}"
+        else:
+            url = f"{self.api_url}/{url}"
+
+        response = await self.http_session.get(url, headers=headers, *args, **kwargs)
+
+        if response.status == 401:
+            if 'www-authenticate' in response.headers:
+                DOZER_LOGGER.info("Reddit token expired when request made, requesting new token and retrying.")
+                await self.get_token()
+                return await self.request(url, headers, *args, **kwargs)
```

Comment:
This passes `headers` as the first argument of `*args`. Is that intentional? The same applies to `TwitchSource`.